kwanghwi/myapp/package.json

```
@@ -0,0 +1,20 @@
+{
+  "name": "myapp",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "start": "nodemon app.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "cors": "^2.8.5",
+    "dotenv": "^16.0.3",
+    "express": "^4.18.2",
+    "morgan": "^1.10.0",
+    "typeorm": "^0.3.11"
+  }
```

package.json 파일의 dependencies와 devDependencies의 차이점을 학습해주세요!

- 개발 환경에서만 사용되는 패키지 모듈은 아래 명령어를 통해 프로젝트에 설치합니다.
`npm install --save-dev "module-name"`

- 반면, 배포 후에도 서버가 구동되기 위해 반드시 필요한 패키지 모듈은 다음과 같이 프로젝트에 설치합니다.
`npm install "module-name"`

- morgan이라는 모듈은 개발 단계에서 log를 확인할 수 있는 모듈기 떄문에 devDependencies에 추가해주는 것이 더 좋을 것 같습니다.

```suggestion
  "dependencies": {
    "cors": "^2.8.5",
    "dotenv": "^16.0.3",
    "express": "^4.18.2",
    "typeorm": "^0.3.11"
  },
  devDependencies: {
    "morgan": "^1.10.0"
  }
```

---

discussion수업할때 종범님께서 
log는 배포 이후에도 서버에서 확인해야하기 때문에 morgan을 dependencies에 둬야 한다고 말씀하셨던걸로 기억하고있는데, 
제가 잘못이해했던건지 확인해주시면 감사하겠습니다!

---

광휘님! 커멘트 감사해요.

배포 후에 백엔드 서버의 로그를 확인할 수 있는 부분은 다양한데요.

말씀하신 대로, morgan을 통해서 배포 후 로그를 확인할 수도 있습니다.

만약 morgan을 통해 배포 이후에도 서버의 로그를 계속 확인하고 싶다면 다음과 같이 적용해볼 수 있겠네요!
```
- package.json의 dependencies 부분에 morgan 추가.
- HTTP 통신을 시도하는 외부 IP 정보, 접속 시간, 통신 결과 등이 포함되어 있는 morgan 로그 option 지정
```

morgan이 제공하는 옵션을 확인해서 배포 이후에도 적용될 수 있도록 수정하면 될 것 같습니다ㅎㅎ